### DISCUSSION TITLE
Best way to troubleshoot issues?

### DISCUSSION THREAD

**Discussion by xstrex** [moralmunky/Home-Assistant-Mail-And-Packages#890] [Q&A]
As a new user to your add-on I'm not entirely sure where to start; everything seems to be configured, but I also see no packages & no mail; so how can i preform some initial troubleshooting to help narrow down the issue? 

If I've followed the provided documentation correctly, and it doesn't work, then what? 
(yes I know this is open-ended, but that's kind of the point, since every integration is slightly different)

**Comment by moralmunky** [comment]
Are you receiving the emails from the carrier you want the information from at the email address you provided the integration?

**Reply by firstof9** [reply]
To expand on this, [check out the wiki](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/wiki/Supported-Shipper-Requirements). That links to the settings to set for USPS, UPS, FedEx as a starting point.

**Reply by xstrex** [reply]
Thank you, and I've already been to and used the wiki to setup my USPS, UPS, FedEx accounts, and they're all emailing to the correct mailbox. I'm also a little vague on how Amazon packages work, but think I've configured that as well. Yet still it seems I only get any kind of notification for USPS packages (not mail), and no other services. I've tried enabling debug logging, but still it doesn't seem to elude to any misconfiguration- It just doesn't work. So what's next? What else can I do to troubleshoot this?

Opening an issue seems like the wrong choice, since it's likely not a bug in the software (until it's proven that it is).

**Reply by xstrex** [reply]
@moralmunky Yep, receiving the emails I would expect from USPS, UPS, FedEx to the email box specified in the integration. It's a mailbox I created (on my server) specifically for this purpose, so everything goes there, and the integration only checks that box. No sub-folders, and it's not used for anything else other than this integration.

**Reply by firstof9** [reply]
> Yet still it seems I only get any kind of notification for USPS packages (not mail)

For incoming regular mail, you need to have setup [USPS informed delivery ](https://www.usps.com/manage/informed-delivery.htm )as well.

**Reply by xstrex** [reply]
> > Yet still it seems I only get any kind of notification for USPS packages (not mail)
> 
> 
> 
> For incoming regular mail, you need to have setup [USPS informed delivery ](https://www.usps.com/manage/informed-delivery.htm )as well.

Understood, and I've done that as well.

**Reply by firstof9** [reply]
The debug output should tell you what it's doing and looking for.

Attach the diagnostics from the integration here, might just be a wonky setting.

![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/1105672/8a4e73cc-53b6-49f6-a061-a848368a619f)

**Reply by xstrex** [reply]
Understood, attaching
[config_entry-mail_and_packages-89385dff6ecf0f8bf1f30aea956d65a5.json](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/files/15195468/config_entry-mail_and_packages-89385dff6ecf0f8bf1f30aea956d65a5.json)

**Reply by xstrex** [reply]
I can also provide a screenshot of the inbox for the mailbox, showing it's contents, if that's helpful.

**Reply by firstof9** [reply]
Your config looks fine, can you attach some debug log as well?

![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/1105672/e5db55bb-09cf-4a25-b5a7-80a5bf8b0f4f)

Let it run for a few mins, then once you hit stop it'll download the log in your browser.

**Reply by xstrex** [reply]
Attached is the debug log, I only redacted my name and non-pertinent information. 

Of course I don't have any packages being delivered today; but I did have 2 delivered yesterday, via amazon, and the package sensors for those packages remained at '0'. 
[home-assistant_mail_and_packages_2024-05-03T14-21-23.741Z.log](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/files/15201314/home-assistant_mail_and_packages_2024-05-03T14-21-23.741Z.log)

**Reply by firstof9** [reply]
```
2024-05-03 08:17:06.847 DEBUG (SyncWorker_3) [custom_components.mail_and_packages.helpers] Informed delivery search date: 03-May-2024
2024-05-03 08:17:06.847 DEBUG (SyncWorker_3) [custom_components.mail_and_packages.helpers] DEBUG subject: Your Daily Digest
2024-05-03 08:17:06.847 DEBUG (SyncWorker_3) [custom_components.mail_and_packages.helpers] DEBUG imap_search: (OR OR OR FROM "USPSInformedDelivery@usps.gov" FROM "USPSInformeddelivery@email.informeddelivery.usps.com" FROM "USPSInformeddelivery@informeddelivery.usps.com" FROM "USPS Informed Delivery" SUBJECT "Your Daily Digest" SINCE 03-May-2024)
2024-05-03 08:17:06.882 DEBUG (SyncWorker_3) [custom_components.mail_and_packages.helpers] email_search value: ('OK', [b''])
```

Shows you haven't received the informed delivery email from USPS, you're not forwarding the USPS emails from gmail are you?

As for Amazon, your amazon sensor would have indicated 1 or 2 packages for yesterday, but those don't get tallied in the main "in transit" sensor as they can sometimes be delivered by USPS/UPS/FedEx/etc.

**Reply by xstrex** [reply]
I am not, mail from USPS is being sent directly to the packages@morphx.net mailbox. 
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/12849470/7e13e280-4575-4477-8ab4-28fafa5b275c)

I understand what you mean by Amazon packages being delivered by other carriers, but wouldn't those carriers sensors be non-zero if that were the case? Yesterday an Amazon delivery driver (actual amazon truck) knocked on my door with a package, and all the package sensors, including the ones for Amazon were all 0. Interestingly the 'mail packages amazon' sensor was '2' on May 1st, but '0' on May 2nd, when the packages were actually delivered. The 'mail packages amazon delivered' sensor has remained '0'. I don't know if this is expected behavior or not, which is why I opened this issue. 
<img width="1443" alt="image" src="https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/12849470/9bf15ca3-8647-4d95-b22c-7659fc7869d6">

**Reply by firstof9** [reply]
> Interestingly the 'mail packages amazon' sensor was '2' on May 1st, but '0' on May 2nd, when the packages were actually delivered.

This is because their emails indicated May 1st as the delivery date.

As for USPS, make sure the setting at the top is enabled:
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/1105672/43f84283-d5a4-4c29-8a67-0a48e06cd0a0)

**Reply by xstrex** [reply]
Yep, Daily Digest emails is set to yes.

**Reply by xstrex** [reply]
Wouldn't the mail amazon package delivered sensor be 2 when the package was actually delivered? Or if it were delayed, wouldn't that of been caught by the exception sensor? Maybe I don't know how this _should_ work.

**Reply by firstof9** [reply]
> wouldn't that of been caught by the exception sensor?

Only if they sent you an exception email would the exceptions increment, but the package counter wouldn't go up as the date didn't change on the original emails. 

> Daily Digest emails is set to yes.

I'm not sure what's going on with your USPS sensors. Can you check to see if the email in your inbox has one of the email addresses:
"USPSInformedDelivery@usps.gov", "USPSInformeddelivery@email.informeddelivery.usps.com", "USPSInformeddelivery@informeddelivery.usps.com"

**Reply by xstrex** [reply]
Ah, that's right. I've got to remember that the integration is only reading emails in the mailbox that it has direct access to. Could you explain the best practice for managing amazon emails? I read that it supports forwarding of emails, but I'm not sure I understand how that works. 

As for USPS; maybe I'm just not getting mail ¯\_(ツ)_/¯ 
I updated the personal settings on the usps site, and ensured the right email address was in there, and saved everything. Got a confirmation email form "USPostalService@usps.com" stating that my personal profile was updated, in my "packages@morphx.net" box, so maybe it's working as expected? Just seems weird that the usps sensors are and are always '0'.

**Reply by xstrex** [reply]
Maybe I spoke too soon, and everything is actually working as expect, and I just don't get a lot of mail. History for 2 weeks.. 
<img width="1228" alt="image" src="https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/12849470/0c41727a-a4bb-46b4-aebb-954c6fcc5939">

**Reply by firstof9** [reply]
Your amazon forwarding seems to be setup fine. Amazon's just not as verbose as other carriers with regards to their email notifications.

Also to note for USPS, junk mail, ie: advertisements get stripped from the count as well.

**Reply by xstrex** [reply]
So maybe it's actually just working as expected? I just don't get that much mail. 

Should I forward _all_ package/shipping related messages from Amazon to my mailbox being monitored by the integration?

**Reply by firstof9** [reply]
That would be the easiest way.
